### PR TITLE
Copter: Send ESC telemetry over mavlink while doing compassmot

### LINKED ISSUE #18874: compassmot stops mavlink telemetry
## Bug report

**Issue details**

When compassmot mode is entered, most mavlink telemetry messages stop

**Version**
github master from today

**Platform**
[  ] All
[  ] AntennaTracker
[ x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
octa-quad-cwx

**Hardware type**
Pixwak4 and CUAV-X7

**How to reproduce using the SITL and mavproxy**

```
sim_vehicle.py -N -v ArduCopter -f octa-quad-cwx -A --uartF=sim:fetteconewireesc --console --map
param set SERIAL5_PROTOCOL 38
param set SERVO_FTW_MASK 255
param set SIM_FTOWESC_ENA 1
reboot
watch  ESC_TELEMETRY_1_TO_4 ESC_TELEMETRY_5_TO_8 ATTITUDE
```
at this point you should see the telemetry messages on the mavlink console

Now start compassmot, the telemetry messages will stop
```
compassmot
```

Also tested it on real hardware with two GCS connected simultaneously to two different UARTs:
- connected mavproxy via USB cable to a Pixhawk4
- and connected QGroundControl via RFD868x to TELEM1 and used the mavlink inspector

As soon as I issued compassmot on mavproxy, all mavlink communication with QGroundControl ceased except `COMPASSMOT_STATUS` and `HEARTBEAT` messages. Those continue to be send.

### PR DESCRIPTION
**Pull request by amilcarlucas**
Fixes #18874

I tested this in real hardware

### CODE REVIEW

**Comment by peterbarker** [comment]
tldr; cause of problem is that we stop running the main loop when doing per-motor compass calibration.  This doesn't fix that, but just sends out the message which someone wants in this specific case.

Correct fix is probably to not stop running the main loop, rather have a state machine like the normal motor test.

I think we should merge this - it's not a big patch and does solve a specific problem.

**Comment by rmackay9** [comment]
I've added a comment re the comment.  After this is modified let's merge
